app/portainer/settings/edge-compute/SettingsEdgeCompute/SettingsEdgeCompute.tsx

```diff
@@ -0,0 +1,128 @@
+import { Formik, Form } from 'formik';
+
+import { Switch } from '@/portainer/components/form-components/SwitchField/Switch';
+import { FormControl } from '@/portainer/components/form-components/FormControl';
+import { Select } from '@/portainer/components/form-components/Input/Select';
+import { Widget, WidgetBody, WidgetTitle } from '@/portainer/components/widget';
+import { LoadingButton } from '@/portainer/components/Button/LoadingButton';
+import { TextTip } from '@/portainer/components/Tip/TextTip';
+
+import styles from './SettingsEdgeCompute.module.css';
+
+export interface FormValues {
+  EdgeAgentCheckinInterval: number;
+  EnableEdgeComputeFeatures: boolean;
+}
+
+interface Props {
+  settings: FormValues;
+  onSubmit(values: FormValues): void;
+}
+
+const checkinIntervalOptions = [
+  {
+    value: 5,
+    label: '5 seconds',
+  },
+  {
+    value: 10,
+    label: '10 seconds',
+  },
+  {
+    value: 30,
+    label: '30 seconds',
+  },
+];
+
+export function SettingsEdgeCompute({ settings, onSubmit }: Props) {
+
+  const initialValues = {
+    EdgeAgentCheckinInterval: settings ? settings.EdgeAgentCheckinInterval : 5,
+    EnableEdgeComputeFeatures: settings ? settings.EnableEdgeComputeFeatures : false,
+  };
+
+  return (
+    <div className="row">
+      <Widget>
+        <WidgetTitle icon="fa-laptop" title="Edge Compute settings" />
+        <WidgetBody>
+          <Formik
+            initialValues={initialValues}
+            enableReinitialize
+            onSubmit={onSubmit}
+            validateOnMount
+          >
+            {({
+              values,
+              errors,
+              handleSubmit,
+              setFieldValue,
+              isSubmitting,
+              isValid,
+              dirty,
+            }) => (
+              <Form
+                className="form-horizontal"
+                onSubmit={handleSubmit}
+                noValidate
+              >
+                <FormControl
+                  inputId="edge_checkin"
+                  label="Edge agent default poll frequency"
+                  tooltip="Interval used by default by each Edge agent to check in with the Portainer instance. Affects Edge environment management and Edge compute features."
+                  errors={errors.EdgeAgentCheckinInterval}
+                >
+                  <Select
+                    value={values.EdgeAgentCheckinInterval}
+                    onChange={(e) =>
+                      setFieldValue(
+                        'EdgeAgentCheckinInterval',
+                        parseInt(e.currentTarget.value, 10)
+                      )
+                    }
+                    options={checkinIntervalOptions}
+                  />
```

Comment:
use formik natives wherever possible.
```suggestion
                  <Field
                    as={Select}
                    value={values.EdgeAgentCheckinInterval}
                    name="EdgeAgentCheckinInterval"
                    options={checkinIntervalOptions}
                  />
```

Reply:
I changed it for the Input fields, the rest didn't work for me, at least for now I'd rather leave it as it is